packages/sdk/src/subgraph.ts

```diff
@@ -152,7 +217,115 @@ export class Subgraph {
         return rxTxs.concat(senderTxs);
       }),
     );
-    this.logger.info({ methodId, methodName, txs }, "Queried active txs");
-    return txs.flat();
+    this.logger.debug({ methodId, methodName, txs }, "Queried active txs");
+    const all = txs.flat();
+
+    // get statuses for txs we are already tracking so we can determine which ones have been fulfilled
+    const allIds = all.map((tx) => tx.txData.transactionId);
+    const unknownTxs = [...this.activeTxs]
+      .filter(([, tx]) => !allIds.includes(tx.txData.transactionId))
+      .map(([, tx]) => tx); // get just the array of txs from the map
+    const unknownPerChain: Record<number, ActiveTransaction[]> = {};
+    unknownTxs.forEach((tx) => {
+      if (!unknownPerChain[tx.txData.receivingChainId]) {
+        unknownPerChain[tx.txData.receivingChainId] = [tx];
+      } else {
+        unknownPerChain[tx.txData.receivingChainId].push(tx);
+      }
+    });
```

Comment:
why does it look at the receiving chain id?

Reply:
Definitely need more comments to highlight the overall thinking/process here. im still not exactly sure what this bit is doing